src/controllers/wallet/atomic-swap/tx-proposal.controller.js

```diff
@@ -53,47 +50,43 @@ async function buildTxProposal(req, res) {
   }
 
   const proposal = partialTx
-    ? PartialTxProposal.fromPartialTx(partialTx, network) : new PartialTxProposal(network);
+    ? PartialTxProposal.fromPartialTx(partialTx, req.wallet.storage)
+    : new PartialTxProposal(req.wallet.storage);
 
   if (sendTokens.utxos && sendTokens.utxos.length > 0) {
     try {
       for (const utxo of sendTokens.utxos) {
-        const txData = req.wallet.getTx(utxo.txId);
+        const txData = await req.wallet.getTx(utxo.txId);
         if (!txData) {
           // utxo not in history
           continue;
         }
         const txout = txData.outputs[utxo.index];
-        if (!oldWallet.canUseUnspentTx(txout, txData.height)) {
+        if (!await transactionUtils.canUseUtxo(
+          { txId: utxo.txId, index: utxo.index },
+          req.wallet.storage,
+        )) {
           // Cannot use this utxo
           continue;
         }
 
-        const addressIndex = req.wallet.getAddressIndex(txout.decoded.address);
-        const addressPath = addressIndex ? req.wallet.getAddressPathForIndex(addressIndex) : '';
+        const addressIndex = await req.wallet.getAddressIndex(txout.decoded.address);
+        const addressPath = addressIndex ? await req.wallet.getAddressPathForIndex(addressIndex) : '';
         let authorities = 0;
-        if (oldWallet.isMintOutput(txout)) {
+        if (transactionUtils.isMint(txout)) {
           authorities += TOKEN_MINT_MASK;
         }
-        if (oldWallet.isMeltOutput(txout)) {
+        if (transactionUtils.isMelt(txout)) {
           authorities += TOKEN_MELT_MASK;
         }
 
-        let tokenId = txout.token;
-        if (!tokenId) {
-          const tokenIndex = oldWallet.getTokenIndex(txout.token_data) - 1;
-          tokenId = txout.token_data === 0
-            ? HATHOR_TOKEN_CONFIG.uid
-            : txData.tx.tokens[tokenIndex].uid;
-        }
-
```

Comment:
Actually `txout.token` is created as a [`tx.to_json_extended`](https://github.com/HathorNetwork/hathor-core/blob/master/hathor/wallet/resources/thin_wallet/address_history.py#L223) which always adds the token, even for HTR.
What it does not add is a tokens array with the token config, so if was never being used and if it somehow entered it would throw an error.